[2024-06-17T04:36:03.332+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-17T04:36:03.395+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-17T04:25:10+00:00 [queued]>
[2024-06-17T04:36:03.414+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-17T04:25:10+00:00 [queued]>
[2024-06-17T04:36:03.415+0000] {taskinstance.py:2306} INFO - Starting attempt 4 of 6
[2024-06-17T04:36:03.441+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): avg_emission_by_year> on 2024-06-17 04:25:10+00:00
[2024-06-17T04:36:03.451+0000] {standard_task_runner.py:63} INFO - Started process 2615 to run task
[2024-06-17T04:36:03.459+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'avg_emission_by_year', 'manual__2024-06-17T04:25:10+00:00', '--job-id', '499', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmp90eqp425']
[2024-06-17T04:36:03.463+0000] {standard_task_runner.py:91} INFO - Job 499: Subtask avg_emission_by_year
[2024-06-17T04:36:03.589+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.avg_emission_by_year manual__2024-06-17T04:25:10+00:00 [running]> on host fa4d239664b0
[2024-06-17T04:36:04.289+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='avg_emission_by_year' AIRFLOW_CTX_EXECUTION_DATE='2024-06-17T04:25:10+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-17T04:25:10+00:00'
[2024-06-17T04:36:04.290+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-17T04:36:04.291+0000] {summary_analysis.py:33} INFO - starting average pollutants emission calculation
[2024-06-17T04:36:04.351+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:36 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-17T04:36:04.985+0000] {summary_analysis.py:45} ERROR - error calculating average emission
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 39, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-17T04:36:04.993+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-17T04:36:04.994+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 39, in avg_emission_by_year
    avg_emision = sub_data.groupby('Year').mean().reset_index()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 9183, in groupby
    return DataFrameGroupBy(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/groupby.py", line 1329, in __init__
    grouper, exclusions, obj = get_grouper(
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/groupby/grouper.py", line 1043, in get_grouper
    raise KeyError(gpr)
KeyError: 'Year'
[2024-06-17T04:36:05.011+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=avg_emission_by_year, run_id=manual__2024-06-17T04:25:10+00:00, execution_date=20240617T042510, start_date=20240617T043603, end_date=20240617T043605
[2024-06-17T04:36:05.038+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 499 for task avg_emission_by_year ('Year'; 2615)
[2024-06-17T04:36:05.082+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-17T04:36:05.165+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-17T04:36:05.171+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
